### ISSUE TITLE
[BUG] rich.progress should fallback to use ─ instead of -

### ISSUE THREAD

**Issue by pcroland** [Textualize/rich#1877]
So I use wsltty mainly where rich.progress uses the `━━` character for the progress bar. I checked it in cmd.exe and it showed `--` for the progress bar. The fallback should be `──` (also used in tables), it would give a connected line.
![img](https://i.kek.sh/4Sy5TIB4O6n.png)

btw: is there any terminal that does not support the `━` character?

**Comment by wasi-master** [comment]
If triage accepts this then I'll open a PR

**Comment by willmcgugan** [comment]
This would would need to handle `options.ascii_only` to default to hyphens.

**Comment by wasi-master** [comment]
Okay thanks

**Comment by wasi-master** [comment]
So the change I mentioned above works, but it has one unexpected side effect. basically options.is_terminal is for checking if the output is a terminal, not specifically command prompt, and because of that the normal behavior also changes. now one thing I notice is that it defaults to ASCII Mode on Command prompt.

So I went on to test if ASCII mode is necessary. and it looks like it isn't since the normal character also works for me, so I checked a bunch of encodings and most of them seem to work fine. now I am a bit stuck on what to do next

![image](https://user-images.githubusercontent.com/63045920/154673712-be26767b-2785-46da-a629-5998dcb00ff0.png)



Before: 
![image](https://user-images.githubusercontent.com/63045920/154670911-22908dba-bd45-4261-ac9b-b8f17928cefd.png)
After:
![image](https://user-images.githubusercontent.com/63045920/154670865-beb990c7-74a9-4c12-bcb8-a2061c2b2714.png)
